artman/cli/converter.py

```diff
@@ -0,0 +1,297 @@
+        args = sys.argv[1:]
+
+    # Get to a normalized set of arguments.
+    try:
+        flags = _parse_args(*args)
+        legacy_config = _load_legacy_config_dict(os.path.abspath(flags.config))
+        new_config = _convert(legacy_config)
+        _write_pb_to_yaml(new_config, flags.output)
+    except Exception:
+        print('Fail to convert `%s`.' % flags.config)
+        raise
+
+
+def _parse_args(*args):
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'config',
+        type=str,
+        help='Path to the legacy artman config')
+    parser.add_argument(
+        '--output',
+        type=str,
+        default=None,
+        help='If specified, the converted yaml will be stored in the given '
+             'output file. Otherwise, the converter will only print out the '
+             'result.')
+    return parser.parse_args(args=args)
+
+
+def _convert(legacy_config):
+    # Compute common section
+    result = Config()
+    if 'common' not in legacy_config:
+        print('`common` field is a required field in the legacy config.')
+        sys.exit(2)
+    legacy_common = legacy_config.get('common')
+    result.common.api_name = legacy_common.get('api_name', 'unspecified')
+    result.common.api_version = legacy_common.get('api_version', 'unspecified')
+    result.common.organization_name = legacy_common.get(
+        'organization_name', 'unspecified')
+
+    # Python dict#get(key, default=value) doesn't well support list-type field.
+    # Instead of using defaultdict, simply do the if checking here.
+    if legacy_common.get('gapic_api_yaml'):
+        result.common.gapic_yaml = _sanitize_repl_var(
+            legacy_common.get('gapic_api_yaml')[0])
+    if legacy_common.get('service_yaml'):
+        result.common.service_yaml = _sanitize_repl_var(
+            legacy_common.get('service_yaml')[0])
+    if legacy_common.get('proto_deps'):
+        result.common.proto_deps.extend(
+            _compute_deps(legacy_common.get('proto_deps')))
+    if legacy_common.get('proto_test_deps'):
+        result.common.test_proto_deps.extend(
+            _compute_deps(legacy_common.get('proto_test_deps')))
+    if legacy_common.get('src_proto_path'):
+        result.common.src_proto_paths.extend(
+            _compute_src_proto_paths(legacy_common.get('src_proto_path')))
+
+    result.artifacts.extend(_compute_artifacts(legacy_config, legacy_common))
+    return result
+
+
+def _compute_artifacts(legacy_config, legacy_common):
+    result = []
+    LANGS = ['java', 'python', 'php', 'ruby', 'go', 'csharp', 'nodejs']
+
+    # Compute artifacts section
+    if legacy_common.get('git_repos'):
+        common_git_repos = legacy_common.get('git_repos')
+    else:
+        common_git_repos = []
+
+    for lang in LANGS:
+        if lang not in legacy_config:
+            continue
+        legacy_artifact_config = legacy_config[lang]
+        if legacy_artifact_config:
+            artifact = Artifact()
```

Comment:
@ethanbao In Python, you can use optional parameters in the constructor, as Luke and I suggested.